src/progpy/models/aircraft_model/small_rotorcraft.py

```diff
@@ -217,27 +217,33 @@ def dx(self, x, u):
 
         # Update state vector
         # -------------------
-        dxdt = np.zeros((len(x),))
 
-        dxdt[0] = vx_a    # x-position increment (airspeed along x-direction)
-        dxdt[1] = vy_a    # y-position increment (airspeed along y-direction)
-        dxdt[2] = vz_a    # z-position increment (airspeed along z-direction)
+        # Positions
+        x['x'] += vx_a*dt
+        x['y'] += vy_a*dt
+        x['z'] += vz_a*dt
 
-        dxdt[3] = p + q * sin_phi * tan_theta + r * cos_phi * tan_theta        # Euler's angle phi increment
-        dxdt[4] = q * cos_phi - r * sin_phi                                    # Euler's angle theta increment
-        dxdt[5] = q * sin_phi / cos_theta + r * cos_phi / cos_theta            # Euler's angle psi increment
+        # Euler's angles
+        x['phi'] += (p + q * sin_phi * tan_theta + r * cos_phi * tan_theta)*dt
+        x['theta'] += (q * cos_phi - r * sin_phi)*dt
+        x['psi'] += (q * sin_phi / cos_theta + r * cos_phi / cos_theta)*dt
 
-        dxdt[6] = ((sin_theta * cos_psi * cos_phi + sin_phi * sin_psi) * T - fe_drag[0]) / m   # Acceleration along x-axis
-        dxdt[7] = ((sin_theta * sin_psi * cos_phi - sin_phi * cos_psi) * T - fe_drag[1]) / m   # Acceleration along y-axis
-        dxdt[8] = -self.parameters['gravity'] + (cos_phi * cos_theta * T - fe_drag[2]) / m   # Acceleration along z-axis
+        # Velocities
+        x['vx'] += dt*((sin_theta * cos_psi * cos_phi + sin_phi * sin_psi) * T - fe_drag[0]) / m
```

Comment:
Should the division by m be included in the parenthesis, so that the entire value is multiplied by dt here? 

I see the same for vy, p, q, and r, so perhaps I'm not thinking about it correctly. If my logic is incorrect, and the division should not be included, then does vz need to be changed to be consistent with this?

Reply:
We discussed this and decided they were functionally the same, so the parenthesis aren't needed